### PR TITLE
Version upgrades for 22Q1

### PR DESCRIPTION
**Pull request by shemnon**
* lots of errorprone fixes
* some license updates
* upgrade the rocksdb version

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: gradle/versions.gradle [hunk 1]
```diff
@@ -15,27 +15,35 @@
...
-    dependencySet(group: 'com.google.errorprone', version: '2.8.0') {
+    dependencySet(group: 'com.google.errorprone', version: '2.10.0') {
       entry 'error_prone_annotation'
       entry 'error_prone_check_api'
       entry 'error_prone_core'
       entry 'error_prone_test_helpers'
     }
 
-    dependency 'com.google.guava:guava:30.1.1-jre'
-    dependency 'com.github.ben-manes.caffeine:caffeine:3.0.4'
```
Comment (matkt) [inline gradle/versions.gradle:37]: we delete because it is no longer used ?

**Review by matkt — APPROVED** [review]
just a question in relation to a dependency otherwise it seems ok to me

File: gradle/versions.gradle (hunk 1 above)
Comment (shemnon) [reply gradle/versions.gradle:37]: Moved to an alpha sort by group.  These were out of order.

**Comment by matkt** [comment]
I think there is an issue related to this commit (upgrade the rocksdb version)

https://github.com/hyperledger/besu/issues/3220

**Comment by shemnon** [comment]
Roll back rocksdb.  I'll be offline for a few days (#marshallfire) so don't wait for my approval.

**Comment by matkt** [comment]
I am no longer with my computer. i will do as soon as possible

**Comment by shemnon** [comment]
https://github.com/hyperledger/besu/pull/3221

**Pull request merged**